src/patch/createPatch.ts

```diff
@@ -41,8 +41,14 @@ function _diff(from: any, to: any, startingPath?: JsonPointer): Operation[] {
 	return operations;
 }
 
-export function diff<T, U>(from: T, to: U): Patch<T, U> {
-	return <Patch<T, U>> createPatch(_diff(from, to));
+export function diff<T>(to: T): Patch<any, T>;
+export function diff<T, U>(from: T, to: U): Patch<T, U>;
+export function diff(from: any, to?: any) {
+	if (!to) {
+		to = from;
+		from = {};
+	}
+	return createPatch(_diff(from, to));
```

Comment:
This confused my slightly, with the switching of the arguments like that...

Wouldn't this be clearer and prevent the need to swap values? I appreciate you would need to update all current consumers of this `diff` function as I am swapping the order of the arguments.

```ts
export function diff<T>(to: T): Patch<any, T>;
export function diff<T, U>(to: T, from: U): Patch<T, U>;
export function diff(to: any, from: any = {}) {
	return createPatch(_diff(from, to));
}
```